### PR TITLE
Add allEntries() in TableEntriesScan

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

add allEntries() in TableEntriesScan

## Brief change log

*(For example:)*
  - *Add the 'scan.startup.mode' configuration to the ArcticScanContext in the flink modules*
  - *The flink 1.12 and 1.14 versions are affected.*
  - *Add the documentation in the flink chapter*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java [hunk 1]
```diff
@@ -205,12 +215,13 @@ public CloseableIterable<IcebergFileEntry> entries() {
   private Table getEntriesTable() {
```
Comment (baiyangtx) [inline core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java]: Maybe should rename to `getMetadataTable`

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/utils/UnKeyedTableUtil.java [hunk 2]
```diff
@@ -60,22 +63,11 @@ public static Snapshot getCurrentSnapshot(UnkeyedTable internalTable) {
...
-        if (status == ManifestEntryFields.Status.ADDED || status == ManifestEntryFields.Status.EXISTING) {
-          GenericRecord dataFile = (GenericRecord) entry.get(ManifestEntryFields.DATA_FILE_ID);
-          String filePath = (String) dataFile.getField(DataFile.FILE_PATH.name());
-          validFilesPath.add(TableFileUtils.getUriPath(filePath));
-        }
-      }
-    } catch (IOException e) {
-      LOG.error("close manifest file error", e);
+    TableEntriesScan manifestReader = TableEntriesScan.builder(internalTable)
+        .includeFileContent(FileContent.DATA, FileContent.POSITION_DELETES)
```
Comment (baiyangtx) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/utils/UnKeyedTableUtil.java:67]: include EQ_DELETE

**Review by baiyangtx — APPROVED** [review]
